### PR TITLE
Fix ssl/t1_trce.c to parse certificate chains with > 1 certificate pr…

### LINKED ISSUE #6994: TLS tracing doesnt work correctly, mis-parses certificate list in handshake
in ssl/t1_trce.c, in ssl_print_extensions, if extslen is zero and there are > 1 certificate in the chain, 0 is
returned which is interpreted as parsing error and ssl_print_certificates errors out. Only tracing is
affected. I tested this in master branch doing TLS 1.3 connections.
Adding a check for extslen == 0 and returning with msgin and msginlen properly updated
makes parsing the entire certificate chain work fine. But cant say that I'm that familiar with
the TLS packet encoding structure so I would prefer someone else take a look.

### PR DESCRIPTION
**Pull request by eforsberg**
…operly

Fixes #6994
CLA: trivial

##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

**Comment by mattcaswell** [comment]
Please can you submit a CLA for this?

https://www.openssl.org/policies/cla.html

File: ssl/t1_trce.c [hunk 1]
```diff
@@ -883,38 +883,43 @@ static int ssl_print_extensions(BIO *bio, int indent, int server,
     size_t extslen, msglen = *msginlen;
     const unsigned char *msg = *msgin;
 
-    BIO_indent(bio, indent, 80);
     if (msglen == 0) {
-        BIO_puts(bio, "No Extensions\n");
```
Comment (mattcaswell) [inline ssl/t1_trce.c]: Why are you removing this?

Comment (eforsberg) [reply ssl/t1_trce.c]: at first glance, seemed rather superfluos output with not much informational value. After
reading up in more detail on TLS packet encoding, I realized this case would only apply
to older TLS/SSL versions before extensions were introduced. Dont remember when that happened. For modern versions, we will instead hit the extslen == 0 case. I'm not opposed to keeping the No extensions printing, but should probably also add it to the extslen == 0 case
but feel it doesnt add much value ?

**Comment by t-j-h** [comment]
It wouldn't matter if CLA: trivial was in place - this change is beyond what I would consider trivial so you will need to send in a CLA.

Thanks,

**Comment by levitte** [comment]
You have all the commit message lines smashed together, so they all become the commit subject line. See `git format-patch -1 --stdout cf40778`

If you insert an empty line between the first two lines of your commit message, the two last lines will become the commit message body, and the CLA bot will have an easier time finding the trivial marker.

**Comment by eforsberg** [comment]
OK, I will email a CLA scan this evening when I can get to my scanner.

File: ssl/t1_trce.c (hunk 1 above)
Comment (mattcaswell) [reply ssl/t1_trce.c]: It's there for the case of an SSLv3 ClientHello which does not have an extensions block.

Comment (mattcaswell) [reply ssl/t1_trce.c]: (so I think it should be put back)

**Comment by levitte** [comment]
Closing and opening to kick the CLA bot

**Comment by eforsberg** [comment]
hmm, git rebase (in this case) seems to have me confused.
Do I make that happen in my local clone first, then push that to my github fork, or can I do it
in github directly on this PR ? Sorry about stupid question, but this is first time using github, usually
I only use git locally.

**Comment by mattcaswell** [comment]
Standing in your local branch type:

````
$ git rebase -i cf40778^
````

This will go into an editor and enable you to move around your commits. Move the two "Fix ssl/t1_trce.c..." commits to the end of the list of all of the commits (keeping the order between them the same). Replace the word "pick" at the beginning of the second one with "f". Now exit the editor.

This will move the commits to be the most recent ones, squash them together, and remove the merge commits. You may have to resolve conflicts. If so then fix any conflicts and then complete the rebase:

````
$ git add ssl/t1_trce.c
$ git rebase --continue
```

Now you can re-push your local branch to github. Assuming your github remote is called "github":

````
$ git push -f github
````

**Comment by InfoHunter** [comment]
Merged as d6c46adf and commit message is also fixed. Closing...

**Pull request closed without merging**